analyzer/processes.go

```diff
@@ -0,0 +1,254 @@
+package analyzer
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+
+	"github.com/brimdata/zed/zio"
+	"golang.org/x/sync/errgroup"
+)
+
+type Operation struct {
+	counter *writeCounter
+	group   *errgroup.Group
+}
+
+func (o *Operation) BytesRead() int64 { return atomic.LoadInt64(&o.counter.written) }
+func (o *Operation) Wait() error      { return o.group.Wait() }
+
+func RunProcesses(ctx context.Context, r io.Reader, confs ...Config) (*Operation, error) {
+	var cmds []*wrappedCmd
+	var writers []io.Writer
+	group, ctx := errgroup.WithContext(ctx)
+	for _, conf := range confs {
+		cmd, writer, err := command(conf)
+		if err != nil {
+			return nil, err
+		}
+		cmds = append(cmds, cmd)
+		writers = append(writers, writer)
+	}
+	writeCounter := new(writeCounter)
+	writers = append(writers, writeCounter)
+	for _, cmd := range cmds {
+		group.Go(cmd.Run)
+	}
+	group.Go(func() error {
+		_, err := io.Copy(io.MultiWriter(writers...), r)
+		for _, w := range writers {
+			if closer, ok := w.(io.Closer); ok {
+				closer.Close()
+			}
+		}
+		// Broken pipe error is a result of a process shutting down. Return nil
+		// here since the process errors are more of interest.
+		if errors.Is(err, syscall.EPIPE) {
+			err = nil
+		}
+		return err
+	})
+	return &Operation{
+		counter: writeCounter,
+		group:   group,
+	}, nil
+}
+
+func command(conf Config) (*wrappedCmd, io.WriteCloser, error) {
+	cmd := exec.Command(conf.Cmd, conf.Args...)
+	cmd.Dir = conf.WorkDir
+	pw, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+	return &wrappedCmd{
+		Cmd:         cmd,
+		stderrPath:  conf.StderrPath,
+		stderrSaver: &prefixSuffixSaver{N: 32 << 10},
+		stdoutPath:  conf.StdoutPath,
+		stdoutSaver: &prefixSuffixSaver{N: 32 << 10},
+	}, pw, nil
+}
+
+type wrappedCmd struct {
+	*exec.Cmd
+	stdoutPath  string
+	stdoutSaver *prefixSuffixSaver
+	stderrPath  string
+	stderrSaver *prefixSuffixSaver
+	wg          sync.WaitGroup
+}
+
+func (c *wrappedCmd) Run() error {
+	stderr, err := stdioWriter(c.stderrPath, c.stderrSaver)
+	if err != nil {
+		return err
+	}
+	defer stderr.Close()
+	stdout, err := stdioWriter(c.stdoutPath, c.stdoutSaver)
+	if err != nil {
+		return err
+	}
+	defer stdout.Close()
+	c.Cmd.Stderr, c.Cmd.Stdout = stderr, stdout
+	err = c.Cmd.Start()
+	if err != nil {
+		return c.error(err)
+	}
+	err = c.Cmd.Wait()
+	return c.error(err)
```

Comment:
Nit: Should be safe to just do `return c.Error(c.Cmd.Wait())` since you're do longer calling `c.Cmd.StdoutPipe` or `c.Cmd.StderrPipe`.

Reply:
Or just `return c.error(c.Cmd.Run())`